### ISSUE TITLE
Cannot mount the generated image

### ISSUE THREAD

**Issue by maaslo** [torarnv/sparsebundlefs#39]
Hello,
I have sparsebundle on btrfs drive shared by samba from openwrt. The samba configuration allows to mark the shared folder as Apple Time Machine Disk. And it was working for my time machine backups. Unfortunately my macbook died and now I want to recover the files from ubuntu.
When I run the script with the debug flag, it just gets frozen and doesn't print anything else than this.

> sparsebundlefs /run/user/1000/gvfs/smb-share\:server\=openwrt.lan\,share\=mac/Tom’s\ MacBook\ Pro.sparsebundle/  /tmp/6 -d -o allow_other
> FUSE library version: 2.9.9
> nullpath_ok: 0
> nopath: 0
> utime_omit_ok: 0
> unique: 2, opcode: INIT (26), nodeid: 0, insize: 56, pid: 0
> INIT: 7.34
> flags=0x33fffffb
> max_readahead=0x00020000
>    INIT: 7.19
>    flags=0x00000011
>    max_readahead=0x00020000
>    max_write=0x00020000
>    max_background=0
>    congestion_threshold=0
>    unique: 2, success, outsize: 40
> unique: 4, opcode: ACCESS (34), nodeid: 1, insize: 48, pid: 1320
>    unique: 4, error: -38 (Function not implemented), outsize: 16
> unique: 6, opcode: LOOKUP (1), nodeid: 1, insize: 47, pid: 1320
> LOOKUP /.Trash
> getattr /.Trash
>    unique: 6, error: -2 (No such file or directory), outsize: 16
> unique: 8, opcode: LOOKUP (1), nodeid: 1, insize: 52, pid: 1320
> LOOKUP /.Trash-1000
> getattr /.Trash-1000
>    unique: 8, error: -2 (No such file or directory), outsize: 16

Without the flags, it finishes with no errors but, I am not able to mount/open the image with hfsplus
>  wrong fs type, bad option, bad superblock on /dev/loop1

Apfs fails with 
> sudo apfs-fuse -l -r pass /tmp/6/sparsebundle.dmg /tmp/apfs/ 
> This doesn't seem to be an apfs volume (invalid superblock).
> Unable to load container.

Parted doesnt give me any information except for sector size

> parted /tmp/4/sparsebundle.dmg print
> WARNING: You are not superuser.  Watch out for permissions.
> Warning: Unable to open /tmp/4/sparsebundle.dmg read-write (Read-only file
> system).  /tmp/4/sparsebundle.dmg has been opened read-only.
> Error: /tmp/4/sparsebundle.dmg: unrecognised disk label
> Model:  (file)                                                            
> Disk /tmp/4/sparsebundle.dmg: 1020GB
> Sector size (logical/physical): 512B/512B
> Partition Table: unknown
> Disk Flags: 

Do you have any idea what could I try to get it mounted?

**Comment by torarnv** [comment]
Was the disk image encrypted?

**Comment by maaslo** [comment]
Yes. But I couldn't decrypt it with the system password, so I gave up on
this.
